packages/jest-config/src/index.ts

```diff
@@ -313,9 +307,23 @@ export function readConfigs(
 
         return true;
       })
-      .map((root, projectIndex) =>
-        readConfig(argv, root, true, configPath, projectIndex),
-      );
+      .map((root, projectIndex) => {
+        // Skip on all except: projectIndex === 0, projects[0] === process.cwd() or realpath(process.cwd()) on Windows, and projects.length === 1
+        const skipArgvConfigOption = !(
```

Comment:
this is hard to read. Could you split it up into parts? e.g.

```ts
const projectIsTheOnlyProject = projectIndex === 0 && projects.length === 1;
const projectIsCwd =
  process.platform === 'win32'
    ? projects[0] === realpath(process.cwd())
    : projects[0] === process.cwd();

const skipArgvConfigOption = !(projectIsTheOnlyProject && projectIsCwd);
```

or something similar?

Reply:
Thanks for pointing it out! I've applied your suggestion :)